### PR TITLE
tests: cover safe_getattr

### PR DESCRIPTION
**Pull request by blueyed**
Via https://github.com/pytest-dev/pytest/pull/6435.

### CODE REVIEW

File: testing/test_compat.py [hunk 1]
```diff
@@ -167,6 +171,8 @@ def test_safe_getattr():
     helper = ErrorsHelper()
     assert safe_getattr(helper, "raise_exception", "default") == "default"
     assert safe_getattr(helper, "raise_fail", "default") == "default"
+    with pytest.raises(BaseException):
+        assert safe_getattr(helper, "raise_baseexception", "default")
```
Comment (RonnyPfannschmidt) [inline testing/test_compat.py]: good find, i recall that we have outcome exception handling - if that's the case can you cover that as well?

Comment (blueyed) [reply testing/test_compat.py]: Not sure what you mean.
I've noticed that it's not covered with https://github.com/pytest-dev/pytest/pull/6435/commits/85a813b9265a7aeb8ccf81b81f5d1585ee3996f7#diff-4ec37b1ad5b7d41fdac56c907d08dde1R335-R340 (reverted there, but still).

Comment (blueyed) [reply testing/test_compat.py]: "outcome exception" is `raise_fail` there (https://github.com/pytest-dev/pytest/pull/6466/files#diff-ad34b016af6f6485ba6c85afe4d9464cR159).

Comment (RonnyPfannschmidt) [reply testing/test_compat.py]: indeed, i missed that bit, thanks for calling out

i think i would have noted it better if it was named  raise_fail_outcome, what do you think about the naming

Comment (blueyed) [reply testing/test_compat.py]: Renamed.

Comment (RonnyPfannschmidt) [reply testing/test_compat.py]: @blueyed thanks, lovely

**Pull request merged**